### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by rsandor**
Attempt to modularize our copy-pasta `error.js` files. Inspired by Navi's `error.js`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  this.report(err);
+};
+
+/**
+ * Reports errors to rollbar. Note this method is automatically bypassed in test
+ * environments (i.e. `process.env.NODE_ENV === 'test'`).
+ * @param {Error} err The error to report.
+ */
+ErrorCat.prototype.report = function (err) {
+  if (this.canUseRollbar()) {
```
Comment (podviaznikov) [inline index.js]: maybe also add `report` flag. https://github.com/CodeNow/api/blob/master/lib/error.js#L90

File: index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    rollbar.handleErrorWithPayloadData(err, { custom: (err.data || {}) }, noop);
```
Comment (podviaznikov) [inline index.js]: `err.data` willl fail if `err` undefined

File: index.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  value: new ErrorCat(),
+  writable: false,
+  enumerable: true
+});
+
+/**
+ * Express error responder middleware.
+ * @type {function}
+ */
+Object.defineProperty(ErrorCat, 'responder', {
```
Comment (podviaznikov) [inline index.js]: Don't really like this name. Too general. Maybe call  it `error-middleware` or something in that direction? Just thoughts

File: package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "error-cat",
+  "version": "0.0.0",
+  "description": "A friendly feline companion that helps you create, track, and report errors.",
+  "main": "index.js",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "test": "jshint index.js && lab -v -c -a code test/*"
```
Comment (podviaznikov) [inline package.json]: maybe create separate commands: `unit` and `lint` and then piper them together?
@bkendall would appreciate

File: index.js (hunk 3 above)
Comment (rsandor) [inline index.js]: What about just `middleware`?

Comment (rsandor) [inline index.js]: Then it would be

``` js
app.use(ErrorCat.middleware);
```

Comment (podviaznikov) [inline index.js]: works for me

File: package.json (hunk 4 above)
Comment (rsandor) [inline package.json]: Can you give me an example of a project that does this?

Comment (podviaznikov) [inline package.json]: https://github.com/CodeNow/api/blob/master/package.json#L133

**Comment by podviaznikov** [comment]
looks good to me

**Comment by bkendall** [comment]
tests!!! <3

**Comment by bkendall** [comment]
:+1: LGTM

**Pull request merged**